doc/gin_files/version_converter.py

```diff
@@ -0,0 +1,212 @@
+import os
+import re
+import sys
+
+from lxml import etree as ET
+
+import odml
+
+try:
+    unicode = unicode
+except NameError:
+    unicode = str
+
+
+class VersionConverter:
+    """
+    Class for converting odml xml files from version 1.0 to 1.1
+    """
+    header = """<?xml version="1.0" encoding="UTF-8"?>\n"""
+
+    _version_map = {
+        'type': 'type'
+    }
+
+    _error_strings = {
+        '<B0>': ''
+    }
+
+    def __init__(self, filename):
+        self.filename = filename
+
+    @staticmethod
+    def convert_odml_file(filename):
+        """
+        Converts a given file to the odml version 1.1.
+        Unites multuple value objects and brings value attributes out of the <value> tag
+        """
+        tree = None
+        if os.path.exists(filename) and os.path.getsize(filename) > 0:
+            VersionConverter.validate_xml(filename)
+            tree = ET.parse(filename)
+            tree = VersionConverter._replace_same_name_entities(tree)
+            root = tree.getroot()
+            root.set("version", "1.1")
+            for prop in root.iter("property"):
+                one_value = True
+                first_value = None
+                for value in prop.iter("value"):
+                    if one_value:
+                        first_value = value
+                        for val_elem in value.iter():
+                            if val_elem.tag != "value" and one_value:
+                                elem_name = VersionConverter._version_map[val_elem.tag] \
+                                    if val_elem.tag in VersionConverter._version_map else val_elem.tag
+                                new_elem = ET.Element(elem_name)
+                                new_elem.text = val_elem.text
+                                value.getparent().append(new_elem)  # appending to the property
+                                value.remove(val_elem) 
+                        one_value = False
+                    else:
+                        first_value.text += ", " + value.text
+                        prop.remove(value)
+        return tree
+
+    @staticmethod
+    def validate_xml(filename):
+        """
+        Fix an xml file by deleting known mismatching tags 
+        """
+        changes_list = []
```

Comment:
Maybe replace the function body with a more condensed code (hope I got it right) ;)

```Python
changes = False
with open(filename, 'r+') as f:
    doc = f.read()
    for k, v in VersionConverter._error_strings.items():
        if k in doc:
            doc = doc.replace(k, VersionConverter._error_strings[k])
            changes = True

    if changes:
        f.truncate(0)
        f.write(doc)
```